apps/builder/app/dashboard/search/search-field.tsx

```diff
@@ -0,0 +1,47 @@
+import { SearchField } from "@webstudio-is/design-system";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+import { dashboardPath } from "~/shared/router-utils";
+
+export const Search = () => {
+  const navigate = useNavigate();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const location = useLocation();
+  const handleCancelSearch = () => {
+    // When user cancels the search, we try to return to the last path they were on.
+    if (location.state?.previousPathname) {
+      return navigate(location.state.previousPathname);
+    }
+    navigate(dashboardPath("projects"));
+  };
+  const isSearchRoute = location.pathname === dashboardPath("search");
+
+  return (
+    <SearchField
+      value={searchParams.get("q") ?? undefined}
+      onChange={(event) => {
+        const value = event.currentTarget.value.trim();
+        if (value === "") {
+          handleCancelSearch();
+          return;
+        }
+        if (isSearchRoute === false) {
+          navigate(
+            {
+              pathname: dashboardPath("search"),
+              search: `?q=${value}`,
+            },
+            // Remember the last path to return to on abort
```

Comment:
on cancel, broke my mind trying to find abort

Reply:
you mean you were searching for onAbort callback but its called onCancel?